src/web/error.rs

```diff
@@ -132,6 +136,110 @@ impl From<PoolError> for IronError {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+#[allow(dead_code)] // FIXME: remove after iron is gone
+pub enum AxumNope {
+    #[error("Requested resource not found")]
+    ResourceNotFound,
+    #[error("Requested build not found")]
+    BuildNotFound,
+    #[error("Requested crate not found")]
+    CrateNotFound,
+    #[error("Requested owner not found")]
+    OwnerNotFound,
+    #[error("Requested crate does not have specified version")]
+    VersionNotFound,
+    #[error("Search yielded no results")]
+    NoResults,
+    #[error("Internal server error")]
+    InternalServerError,
+    #[error("internal error")]
+    InternalError(#[from] anyhow::Error),
+    #[error("internal threading error")]
+    InternalTokio(#[from] tokio::task::JoinError),
```

Comment:
This would give me: 
```
error[E0277]: `?` couldn't convert the error to `AxumNope`
  --> src/web/sitemap.rs:94:11
   |
94 |     .await??;
   |           ^ the trait `std::convert::From<JoinError>` is not implemented for `AxumNope`
   |
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
   = help: the trait `std::convert::From<anyhow::Error>` is implemented for `AxumNope`
   = note: required for `Result<_, AxumNope>` to implement `FromResidual<Result<Infallible, JoinError>>`
```

I assumed because of similar reasons as for the task join error. 

Or am I missing something?

Reply:
`.await.context("failed to join thread")??` should work, that converts it to anyhow::Error before converting to AxumNope.

Reply:
hm.. I somehow missed that. 

It works like this, thanks!